### PR TITLE
docs: add security.md

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

Please fill in the fields below to submit a pull request.  The more information that is provided, the better.

Fixes https://github.com/theupdateframework/go-tuf/issues/371
Release Notes: 

**Types of changes**:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

**Description of the changes being introduced by the pull request**:

**Please verify and check that the pull request fulfills the following requirements**:

- [ ] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Review by trishankatdatadog — DISMISSED** [review]

**Comment by trishankatdatadog** [comment]
LGTM, Asra, including the Google Form. Thanks so much for doing this!

File: docs/SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+## Supported Versions
+
+You may report issues for the most recent version of go-tuf. We will not retroactively make changes to older versions.
+
+## Reporting a Vulnerability
+
+If you discover a potential security issue in this project we ask that you notify the go-tuf maintainers via our [vulnerability reporting form](https://forms.gle/ShM4s3mLbUAx5QHo8). At the minimum, the report must contain the following:
+* A description of the issue.
+* Instructions to reproduce the issue.
+
```
[the remark this answers is not in the document]
Comment (asraa) [reply docs/SECURITY.md:14]: Clarified!

**Review by rdimitrov — APPROVED** [review]

**Review by znewman01 — DISMISSED** [review]

File: docs/SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+You may report issues for the most recent version of go-tuf. We will not retroactively make changes to older versions.
+
+## Reporting a Vulnerability
+
+If you discover a potential security issue in this project we ask that you notify the go-tuf maintainers via our [vulnerability reporting form](https://forms.gle/ShM4s3mLbUAx5QHo8). At the minimum, the report must contain the following:
+* A description of the issue.
+* A specific version or commit SHA of `go-tuf` where the issue reproduces.
+* Instructions to reproduce the issue.
+
+Please do **not** create a public GitHub issue to submit vulnerability reports. The GitHub issue tracker is intended for bug reports and feature requests. Major feature requests, such as design changes to the specification, should be proposed via a [TUF Augmentation Protocol](https://theupdateframework.github.io/specification/latest/#tuf-augmentation-proposal-tap-support) (TAP).
```
Comment (trishankatdatadog) [inline docs/SECURITY.md:16]:
```suggestion
Please do **not** create a public GitHub issue or pull request to submit vulnerability reports. These public trackers are intended for non-time-sensitive bug reports and feature requests. Major feature requests, such as design changes to the specification, should be proposed via a [TUF Augmentation Protocol](https://theupdateframework.github.io/specification/latest/#tuf-augmentation-proposal-tap-support) (TAP).
```

**Review by trishankatdatadog — CHANGES_REQUESTED** [review]

**Review by trishankatdatadog — DISMISSED** [review]

**Review by rdimitrov — COMMENTED** [review]
Let's resolve the DCO failure and I think we should be okay 👍

**Comment by asraa** [comment]
SG! I really don't know what happened with that DCO failure. Maybe trishank had pushed some commit with co-author?

**Review by asraa — COMMENTED** [review]
Fixed DCO, just squashed the commit onto master

**Review by trishankatdatadog — APPROVED** [review]

**Pull request merged**